tutor/src/screens/course-settings/delete-course-button.js

```diff
@@ -0,0 +1,98 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { observer } from 'mobx-react';
+import { observable, action } from 'mobx';
+import { Modal, Button, OverlayTrigger, Tooltip } from 'react-bootstrap';
+import { isEmpty } from 'lodash';
+import { Icon } from 'shared';
+import Course from '../../models/course';
+
+export default
+@observer
+class DeleteCourseModal extends React.Component {
+
+  static propTypes = {
+    course: PropTypes.instanceOf(Course).isRequired,
+    history: PropTypes.shape({
+      push: PropTypes.func,
+    }).isRequired,
+  }
+
+  @observable showModal = false;
+
+  @action.bound close() {
+    this.showModal = false;
+  }
+
+  @action.bound open() {
+    this.showModal = true;
+  }
+
+  @action.bound deleteCourse() {
+    const { course, history } = this.props;
+    const courseTeacherMembership = course.currentCourseTeacher;
+    if (courseTeacherMembership.isTeacherOfCourse) {
+      courseTeacherMembership.drop().then(history.push('/dashboard'));
+    }
```

Comment:
I was still seeing a test course on the dashboard after deleting it. Not sure if we need to refetch or if there's a computed property that isn't updating, hmm.

Reply:
I merged the `my-courses` branch and it is working for me. @jivey